geoapps/edge_detection/application.py

```diff
@@ -223,148 +223,54 @@ def update_name(self, _):
-                                y[i_min:i_max, j_min:j_max][coord[:, 1], coord[:, 0]],
-                                z[i_min:i_max, j_min:j_max][coord[:, 1], coord[:, 0]],
-                            ]
-                        )
-            if coords:
-                coord = np.vstack(coords)
-                self.objects.lines = coord
-                self.plot_store_lines()
-            else:
-                self.objects.lines = None
-
-    def plot_store_lines(self):
-
-        xy = self.objects.lines
-        indices_1 = filter_xy(
-            xy[1::2, 0],
-            xy[1::2, 1],
-            self.resolution.value,
-            window={
-                "center": [
-                    self.window_center_x.value,
-                    self.window_center_y.value,
-                ],
-                "size": [
-                    self.window_width.value,
-                    self.window_height.value,
-                ],
-                "azimuth": self.window_azimuth.value,
-            },
-        )
-        indices_2 = filter_xy(
-            xy[::2, 0],
-            xy[::2, 1],
-            self.resolution.value,
-            window={
-                "center": [
-                    self.window_center_x.value,
-                    self.window_center_y.value,
-                ],
-                "size": [
-                    self.window_width.value,
-                    self.window_height.value,
-                ],
-                "azimuth": self.window_azimuth.value,
-            },
+        _, _, xy = EdgeDetectionDriver.get_edges(
+            new_params.objects,
+            new_params.data,
+            new_params.sigma,
+            new_params.line_length,
+            new_params.threshold,
+            new_params.line_gap,
+            new_params.window_size,
+            new_params.window_center_x,
+            new_params.window_center_y,
+            new_params.window_width,
+            new_params.window_height,
+            new_params.window_azimuth,
+            new_params.resolution,
         )
 
-        indices = np.kron(
-            np.any(np.c_[indices_1, indices_2], axis=1),
-            np.ones(2),
-        ).astype(bool)
-
-        xy = self.objects.lines[indices, :2]
         self.collections = [
             collections.LineCollection(
                 np.reshape(xy, (-1, 2, 2)), colors="k", linewidths=2
             )
         ]
-        self.refresh.value = False
-        self.refresh.value = True  # Trigger refresh
-
-        if np.any(xy):
-            vertices = np.vstack(self.objects.lines[indices, :])
-            cells = np.arange(vertices.shape[0]).astype("uint32").reshape((-1, 2))
-            if np.any(cells):
-                self.trigger.vertices = vertices
-                self.trigger.cells = cells
-        else:
-            self.trigger.vertices = None
-            self.trigger.cells = None
+        self.refresh.value = True
+
+    def get_param_dict(self):
```

Comment:
Great, and looks pretty general. Can we move this to the base Params class? See if other applications can re-use it.